File: Divide Array Into Equal Pairs/Divide Array Into Equal Pairs.py
# Solution for Divide Array Into Equal Pairs in PY

class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        mp = {}
        for num in nums:
            mp[num] = mp.get(num, 0) + 1
        for it in mp:
            if mp[it] % 2 != 0:
                return False
        return True

File: Divide Array Into Equal Pairs/Divide Array Into Equal Pairs.cpp
// Solution for Divide Array Into Equal Pairs in CPP

class Solution {
public:
    bool divideArray(vector<int>& nums) {
        unordered_map<int, int> mp;
        for (const auto& num: nums) mp[num]++;
        for (const auto& it: mp) {
            if (it.second % 2 != 0) return false;
        }
        return true;
    }
};
